external-providers/generic-external-provider/pkg/server_configurations/generic/service_client_test.go

```diff
@@ -0,0 +1,169 @@
+package generic_test
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/bombsimon/logrusr/v3"
+	"github.com/getkin/kin-openapi/openapi3gen"
+	"github.com/konveyor/analyzer-lsp/lsp/protocol"
+	"github.com/konveyor/analyzer-lsp/provider"
+	"github.com/konveyor/generic-external-provider/pkg/generic_external_provider"
+	"github.com/konveyor/generic-external-provider/pkg/server_configurations/generic"
+	"gopkg.in/yaml.v2"
+
+	"github.com/sirupsen/logrus"
+)
+
+// For debugging
+type EvaluateCall struct {
+	ServiceClient provider.ServiceClient
+	Cap           string
+	ConditionInfo []byte
+}
+
+func TestHopefullyNothingBroke(t *testing.T) {
+	flag.Parse()
+	logrusLog := logrus.New()
+	logrusLog.SetOutput(os.Stdout)
+	logrusLog.SetFormatter(&logrus.TextFormatter{})
+	logrusLog.SetLevel(logrus.Level(5))
+	log := logrusr.New(logrusLog)
+	ctx := context.TODO()
+
+	prov := generic_external_provider.NewGenericProvider("generic")
+
+	goplsSC, err := prov.Init(ctx, log, provider.InitConfig{
+		ProviderSpecificConfig: map[string]interface{}{
+			"lspServerName":    "generic",
+			"lspServerPath":    "gopls",
+			"lspServerArgs":    []interface{}{"-vv", "-logfile", "debug-go.log", "-rpc.trace"},
+			"workspaceFolders": []interface{}{"file:///home/jonah/Projects/analyzer-lsp/examples/golang"},
+		},
+	})
+	_ = goplsSC
+
+	if err != nil {
+		fmt.Printf("%v\n", err)
+		panic(err)
+	}
+
+	var res json.RawMessage
+
+	p0 := protocol.WorkspaceSymbolParams{
+		Query: "dummy.HelloWorld",
+	}
+
+	err = goplsSC.(*generic.GenericServiceClient).Conn.Call(ctx, "workspace/symbol", p0).Await(ctx, &res)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("%s\n", string(res))
+
+	var calls []EvaluateCall
+	calls = append(
+		calls,
+		// EvaluateCall{pythonSC, "referenced", []byte(`{"referenced":{pattern: "def hello_world"}}`)},
```

Comment:
should these be removed?

Reply:
For now I'm good with removing them. At a later date we'll probably need some proper tests written up